c++/src/ColumnPrinter.cc

```diff
@@ -252,7 +256,7 @@ namespace orc {
       break;
 
     case STRUCT:
-      result = new StructColumnPrinter(buffer, type);
+      result = new StructColumnPrinter(buffer, type, selectedColumns);
```

Comment:
For example, if an ORC file contains columns INT, STRUCT<STRING, BOOLEAN>, running
./file-contents --columns=2 file.orc
will select the STRUCT column. If we allowed selection of subcolumns, then it is unclear which column the above command will select: STRUCT or STRING.